internal/pkg/v2/infrastructure/redis/device_service.go

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package redis
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+	model "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	"github.com/gomodule/redigo/redis"
+)
+
+const DeviceServiceCollection = "v2:deviceService"
+
+// addDeviceService adds a new device service into DB
+func addDeviceService(conn redis.Conn, ds model.DeviceService) (addedDeviceService model.DeviceService, edgeXerr errors.EdgeX) {
+	// retrieve Device Service by Id first to ensure there is no Id conflict; when Id exists, return duplicate error
+	_, edgeXerr = deviceServiceById(conn, ds.Id)
```

Comment:
use EXISITS command like you do for HEXISTS for name

Reply:
Updated.